File: src/android/app/src/main/java/org/citra/citra_emu/features/settings/model/view/SubmenuSetting.java
package org.citra.citra_emu.features.settings.model.view;

import org.citra.citra_emu.features.settings.model.Setting;

public final class SubmenuSetting extends SettingsItem {
    private String mMenuKey;

    public SubmenuSetting(String key, Setting setting, int titleId, int descriptionId, String menuKey) {
        super(key, null, setting, titleId, descriptionId);
        mMenuKey = menuKey;
    }

    public String getMenuKey() {
        return mMenuKey;
    }

    @Override
    public int getType() {
        return TYPE_SUBMENU;
    }
}

File: src/android/app/src/main/java/org/citra/citra_emu/features/settings/model/view/SubmenuSetting.kt
// Copyright 2023 Citra Emulator Project
// Licensed under GPLv2 or any later version
// Refer to the license.txt file included.

package org.citra.citra_emu.features.settings.model.view

class SubmenuSetting(
    titleId: Int,
    descriptionId: Int,
    val menuKey: String
) : SettingsItem(null, titleId, descriptionId) {
    override val type = TYPE_SUBMENU
}
